components/renku_data_services/project/db.py

```diff
@@ -173,18 +168,37 @@ async def insert_project(
         self, session: AsyncSession, user: base_models.APIUser, project: models.Project
     ) -> models.Project:
         """Insert a new project entry."""
-        project_orm = schemas.ProjectORM.load(project)
-        project_orm.creation_date = datetime.now(timezone.utc).replace(microsecond=0)
-        project_orm.created_by = user.id
+        ns, _ = await self.group_repo.get_ns_group_orm(user, project.namespace)
+        if not ns:
+            raise errors.MissingResourceError(
+                message=f"The project cannot be created because the namespace {project.namespace} does not exist"
+            )
+        user_id = cast(str, user.id)
+        repos = [schemas.ProjectRepositoryORM(url) for url in (project.repositories or [])]
+        slug = project.slug or base_models.Slug.from_name(project.name).value
+        project_orm = schemas.ProjectORM(
+            name=project.name,
+            visibility=models.Visibility(project.visibility)
+            if isinstance(project.visibility, str)
+            else project.visibility,
+            created_by_id=user_id,
+            description=project.description,
+            ltst_prj_slug=schemas.ProjectSlug(slug, ltst_ns_slug=ns.ltst_ns_slug or ns),
+            repositories=repos,
+            creation_date=datetime.now(timezone.utc).replace(microsecond=0),
+        )
         session.add(project_orm)
+        await session.flush()
 
-        project = project_orm.dump()
-        public_project = project.visibility == Visibility.public
-        if project.id is None:
+        project_dump = project_orm.dump()
+        public_project = project_dump.visibility == Visibility.public
+        if project_dump.id is None:
             raise errors.BaseError(detail="The created project does not have an ID but it should.")
-        await self.project_authz.create_project(requested_by=user, project_id=project.id, public_project=public_project)
+        await self.project_authz.create_project(
+            requested_by=user, project_id=project_dump.id, public_project=public_project
+        )
 
-        return project
+        return project_dump
 
     @with_db_transaction
     @dispatch_message(create_project_update_message)
```

Comment:
we'll probably need to send the slug to search as well, but can create an issue to do so

Reply:
https://github.com/SwissDataScienceCenter/renku-data-services/issues/160